test/inductor/test_torchinductor.py

```diff
@@ -6769,7 +6769,7 @@ def fn(x1, x2):
                 traced = make_fx(fn)(x1, x2)
                 compiled = compile_fx_inner(traced, [x1, x2])
                 assert same(fn(x1, x2)[0], compiled([x1, x2])[0], equal_nan=True)
-                assert metrics.generated_cpp_vec_kernel_count == 1
+                assert metrics.generated_cpp_vec_kernel_count == 2
```

Comment:
Why did this change?

Reply:
This is because the tail kernel of the tile2d is also vectorized along the outer loop index (i0 below). Originally it is a scalar kernel. The non-contiguous load falls back to the scalar inner loop (tmp11 below) and other code is vectorized as much as possible. Therefore, we have both tile2d main kernel and tile2d tail kernel vectorized, which is counted as 2 now.

Generated tail kernel of tile2d below:
```c++
                #pragma GCC ivdep
                for(long i1=static_cast<long>(0); i1<static_cast<long>(10); i1+=static_cast<long>(1))
                {
                    auto tmp0 = at::vec::Vectorized<float>::loadu(in_ptr0 + static_cast<long>((16*i0) + (20*i1)));
                    auto tmp11 = ([&]() { __at_align__ float tmpbuf[16]; for (long i0_inner = 0; i0_inner < 16; i0_inner++) tmpbuf[i0_inner] = in_ptr1[static_cast<long>(i1 + (10*i0_inner) + (160*i0))]; return at::vec::Vectorized<float>::loadu(tmpbuf); })();
                    auto tmp1 = tmp0.abs();
                    auto tmp2 = tmp1.sin();
                    auto tmp3 = tmp2.neg();
                    auto tmp4 = tmp3 * tmp3;
                    auto tmp5 = decltype(tmp4)(1)/(decltype(tmp4)(1) + tmp4.neg().exp());
                    auto tmp6 = at::vec::clamp_min(tmp5, decltype(tmp5)(0));
                    auto tmp7 = tmp6.cos();
                    auto tmp8 = tmp7.exp();
                    auto tmp9 = tmp8.sqrt();
                    auto tmp10 = tmp9 + tmp0;
                    auto tmp12 = tmp10 - tmp11;
                    auto tmp13 = tmp12 * tmp0;
                    auto tmp14 = tmp13 / tmp0;
                    auto tmp15 = tmp14 * tmp14;
                    auto tmp16 = tmp15 * tmp15;
                    auto tmp17 = tmp16 * tmp14;
                    auto tmp18 = tmp17 * tmp17;
                    auto tmp19 = tmp18.log();
                    auto tmp20 = tmp19.floor();
                    auto tmp21 = tmp20.ceil();
                    auto tmp22 = tmp21.trunc();
                    auto tmp23 = tmp22.lgamma();
                    auto tmp24 = tmp23.fmod(tmp11);
                    auto tmp25 = decltype(tmp24)::blendv(decltype(tmp24)(0), decltype(tmp24)(1), decltype(tmp24)(0) < tmp24);
                    auto tmp26 = decltype(tmp24)::blendv(decltype(tmp24)(0), decltype(tmp24)(1), tmp24 < decltype(tmp24)(0));
                    auto tmp27 = tmp25 - tmp26;
                    auto tmp28 = tmp27 + tmp11;
                    tmp28.store(out_ptr0 + static_cast<long>((16*i0) + (20*i1)));
                }
```